hangups/conversation.py

```diff
@@ -805,43 +805,77 @@ def _add_conversation(self, conversation, events=[]):
 
     @asyncio.coroutine
     def _on_state_update(self, state_update):
-        """Receive a StateUpdate and fan out to Conversations."""
+        """Receive a StateUpdate and fan out to Conversations.
+
+        Args:
+            state_update: hangouts_pb2.StateUpdate instance
+        """
         # Handle updating a conversation
         if state_update.HasField('conversation'):
             self._handle_conversation(state_update.conversation)
```

Comment:
There might be a bug here (which was in the existing code as well). If the state update contains the conversation, `_handle_conversation` will add that conversation to the `ConversationList` if it's unknown. However, the conversation in the state update is actually only a delta (see comment in `update_conversation`). This would cause the delta to be used as the complete conversation, and prevent the complete conversation from being fetched.